scvi/train/_trainingplans.py

```diff
@@ -124,6 +125,19 @@ def __init__(
         if "kl_weight" in self._loss_args:
             self.loss_kwargs.update({"kl_weight": self.kl_weight})
 
+        self.train_running_metrics = dict(
+            elbo=MeanMetric(),
+            reconstruction_loss=MeanMetric(),
+            kl_local=MeanMetric(),
+            kl_global=MeanMetric(),
+        )
+        self.val_running_metrics = dict(
+            elbo=MeanMetric(),
+            reconstruction_loss=MeanMetric(),
+            kl_local=MeanMetric(),
+            kl_global=MeanMetric(),
+        )
+
```

Comment:
let's talk about this in person, I think we need to implement a custom metric object, which will even further reduce the custom methods you added.

Reply:
well actually, I think we just write `ElboMetric` as a subclass of their object, and it keeps track of the running metrics and the `compute` method would return a dictionary of elbo and it's components.

Reply:
Sounds good, this will also be easier if we want to include custom metrics natively